flink-yarn-tests/src/test/java/org/apache/flink/yarn/testjob/YarnTestCacheJob.java

```diff
@@ -44,11 +44,16 @@
 public class YarnTestCacheJob {
 	private static final List<String> LIST = ImmutableList.of("test1", "test2");
 
-	public static JobGraph getDistributedCacheJobGraph() {
+	public static JobGraph getDistributedCacheJobGraph() throws IOException {
 		final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
 
-		String cacheFilePath = Thread.currentThread().getContextClassLoader().getResource("cache.properties").getFile();
-
+		final String cacheFilePath = Thread.currentThread().getContextClassLoader()
+			.getResource("cache.properties").getFile();
+		final File testDirectory = File.createTempFile("test_directory", "");
```

Comment:
Would it make sense use instead a `TemporaryFolder` rule in `YarnDistributedCacheITCase` and pass in the test directory to the `getDistributedCacheJobGraph(File testDirectory)`?

Reply:
I will add a new argument for `getDistributedCacheJobGraph`.